lib/java/os-environment/src/main/java/org/enso/os/environment/jni/Channel.java

```diff
@@ -0,0 +1,315 @@
+    @SuppressWarnings("unchecked")
+    var factory =
+        (Supplier<Persistance.Pool>) Class.forName(poolClassName).getConstructor().newInstance();
+    var pool = factory.get();
+    var channel = new Channel(id, pool, threadId, callbackFn);
+    var prev = ID_TO_CHANNEL.put(id, channel);
+    return prev == null;
+  }
+
+  /**
+   * <em>Executes a message</em> in the other JVM. The message is any subclass of {@link Function}
+   * registered for persistance via {@link Persistable @Persistable} annotation into the {@link
+   * Persistance.Pool pool associated with this JVM}. The result (which is of type {@code R}) also
+   * has to be registered for serde.
+   *
+   * <p>
+   *
+   * @param resultType class with the type of {@code R} to use for deserialization
+   * @param msg the message that gets serialized, transferred into the other JVM, deserialized on
+   *     the other side and {@link Message#evaluate() evaluated} there
+   * @param <R> the type of result we expect the message to return
+   * @return the value gets computed via {@link Message#evaluate()} in the other JVM and then it
+   *     gets serialized and transferred back to us. Deserialized and the value is then returned
+   *     from this method
+   */
+  public final <R> R execute(Class<R> resultType, Function<Channel, R> msg) {
+    if (this.isolate == -1) {
+      return executeImpl(pool, resultType, msg, this::toHotSpotMessage);
+    } else {
+      return executeImpl(pool, resultType, msg, this::toSubstrateMessage);
+    }
+  }
+
+  private static final CEntryPointLiteral<CFunctionPointer> CALLBACK_FN =
+      CEntryPointLiteral.create(
+          Channel.class,
+          "acceptRequestFromHotSpotJvm",
+          IsolateThread.class,
+          long.class,
+          CCharPointer.class,
+          long.class);
+
+  @CEntryPoint
+  private static long acceptRequestFromHotSpotJvm(
+      IsolateThread threadId, long id, CCharPointer data, long size) throws Throwable {
+
+    var channel = ID_TO_CHANNEL.get(id);
+    assert channel != null : "There must be a channel " + id + " but " + ID_TO_CHANNEL;
+    try {
+      var len = handleWithChannel(channel, data.rawValue(), size);
+      return len;
+    } catch (Throwable ex) {
+      channel.printStackTrace(ex, true);
+      var exceptionMessage =
+          ex.getMessage()
+              .subSequence(0, Math.min(ex.getMessage().length(), (int) Math.min(2048, size)));
+      CTypeConversion.toCString(exceptionMessage, data, WordFactory.unsigned(size));
+      return -2L;
+    }
+  }
+
+  private static long handleWithChannel(Channel channel, long address, long size) throws Throwable {
+    var seg = MemorySegment.ofAddress(address).reinterpret(size);
+    var buf = seg.asByteBuffer();
+    var ref = channel.pool.read(buf, null);
+    var msg = ref.get(Function.class);
+    @SuppressWarnings("unchecked")
+    var res = msg.apply(channel);
+    var bytes = Persistables.POOL.write(res, null);
```

Comment:
Where is the `Persistables` class comming from? Where should it be generated? I cannot find it anywhere after compilation.

Reply:
Generated by `persistance-dsl` annotation processor.